scipy/sparse/tests/test_base.py

```diff
@@ -2982,6 +2982,29 @@ def test_fancy_indexing_randomized(self):
         assert_raises(IndexError, S.__getitem__, (I_bad,J))
         assert_raises(IndexError, S.__getitem__, (I,J_bad))
 
+    def test_missized_masking(self):
+        np.random.seed(1234)  # make runs repeatable
+        rng = np.random.default_rng(1234)
```

Comment:
```suggestion
        rng = np.random.default_rng(1234)  # make runs repeatable
```
But I would prefer if this didn't use random values at all.
We can't tell where the True values occur when written this way.
And the following test suggests that it (at least used to) matter where the mask values are.

Can we just set e.g. `row_long` to a specific set of 0/1 values?
`row_long = np.array([0, 1, 1, 0, 1, 0], dtype=bool)` or maybe just `row_long = np.zeros(6, dtype=bool)`
Then we at least know what we are testing.

Reply:
I can do that. I can find it a little hard to tell what the length is of arrays written out like that, so I've gone with a little function to generate a known pattern.

Does that work for you?

Reply:
+1 on avoiding a list of numbers.
But the specific form of alternating suggests we are testing that aspect.
How about `np.ones(6, dtype=bool)`?

Reply:
:+1: